src/features/selectedPost.ts

```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+
+const initialState = null as Post | null;
+
+export const selectedPostSlice = createSlice({
+  name: 'selectedPost',
+  initialState,
+  reducers: {
+    setSelectedPost: (_, actions) => actions.payload,
```

Comment:
Consider renaming 'actions' to 'action' in the reducer. This would follow the convention that 'action' is a single dispatched event, while 'actions' would imply a collection of dispatched events, which is not the case here.